src/lib/loadInstructions.js

```diff
@@ -10,47 +21,116 @@ function readYamlFile(p) {
   return yaml.load(txt);
 }
 
+/*
+Parse a match bit string (e.g., "0100000----------101-----0010011")
+into an array of single characters, each representing one bit.
+
+Returns null if the string isn’t 16 or 32 bits long.
+*/
 function parseMatchBits(matchStr) {
-  if (!matchStr || matchStr.length !== 32) return null;
+  if (!matchStr) return null;
+  if (matchStr.length !== 32 && matchStr.length !== 16) return null;
```

Comment:
Does it matter if the length is not 32 or 16? It seems confusing to allow this function to return `null` sometimes and not others. The API could be simpler if it always returned a "valid" value for all inputs. If, on the other hand, we want it to be an _error_ to pass non-32-or-16-bit input, then a simple `console.assert()` would be a better way to noisily signal an error when something has gone wrong.